pkg/webhook/mutation.go

```diff
@@ -162,9 +166,47 @@ func (h *mutationHandler) Handle(ctx context.Context, req admission.Request) adm
 }
 
 func (h *mutationHandler) mutateRequest(ctx context.Context, req admission.Request) (admission.Response, error) {
-	// TODO: place mutation logic here
-	mutatedObject := req.Object
-	resp := admission.PatchResponseFromRaw(req.Object.Raw, mutatedObject.Raw)
+	okResp := admission.Response{
+		AdmissionResponse: admissionv1beta1.AdmissionResponse{
+			Allowed: true,
+			Result: &metav1.Status{
+				Code: int32(http.StatusOK),
+			},
+		},
+	}
+	ns := &corev1.Namespace{}
+	if err := h.client.Get(ctx, types.NamespacedName{Name: req.AdmissionRequest.Namespace}, ns); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			log.Info("Namespace not found", "name", req.AdmissionRequest.Namespace)
+			return okResp, nil
+		}
+		// bypass cached client and ask api-server directly
+		err = h.reader.Get(ctx, types.NamespacedName{Name: req.AdmissionRequest.Namespace}, ns)
+		if err != nil {
+			log.Info("Namespace not found", "name", req.AdmissionRequest.Namespace)
+			return okResp, nil
+		}
+	}
+	obj := unstructured.Unstructured{}
+	err := obj.UnmarshalJSON(req.Object.Raw)
+	if err != nil {
+		log.Error(err, "Failed to unmarshal", "object", string(req.Object.Raw))
+		return okResp, nil
+	}
+
+	err = h.mutationSystem.Mutate(&obj, ns)
```

Comment:
I wonder if we should return a flag that says if we mutated or not the object and not create the patch in that case

Reply:
Two ways of doing this. We can either clone the obj before mutating and compare the two, or have the mutation function return whether anything was mutated, and bubble it up to here. 
Maybe a new PR so that we can have the basic mutation working (especially if we take the second approach)?

Reply:
What is the goal of doing this? To reduce the latency of serialization/deserialization?

Reply:
I'd like to be on the safe side, and if there's nothing to do not going through the serialization / deserialization / patch if the object we are addressing is the same.

Reply:
ack, SGTM

Reply:
Let me open a new PR on this.

Reply:
#1026